check/checksum_test.go

```diff
@@ -0,0 +1,188 @@
+package check
+
+import (
+	"crypto/md5"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+func checksumTestdata(t *testing.T) string {
+	out := t.TempDir()
+
+	src := filepath.Join("..", "testdata")
+	ls, err := os.ReadDir(src)
+	if err != nil {
+		t.Fatalf("expected no error reading testdata directory, got %s", err)
+	}
+
+	for _, f := range ls {
+		r, err := os.Open(filepath.Join(src, f.Name()))
+		if err != nil {
+			t.Fatalf("expected no error opening %s in testdata directory, got %s", f.Name(), err)
+		}
+
+		w, err := os.Create(filepath.Join(out, f.Name()))
+		if err != nil {
+			t.Fatalf("expected no error creating %s in tmp testdata directory, got %s", f.Name(), err)
+		}
+
+		if _, err := io.Copy(w, r); err != nil {
+			t.Fatalf("expected no error writing %s, got %s", f.Name(), err)
+		}
+	}
+
+	return out
+}
+
+func TestCreate(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		src      string
+		expected int
+		err      bool
+	}{
+		{
+			"Src directory does not exists",
+			"../no-dir",
```

Comment:
Talvez isso dê erro em Windows?

```suggestion
			filepath.Join("..", "no-dir"),
```